File: ios/LaneShadow/Sandbox/Stories/LSDisplayStories.swift
import LaneShadowTheme
import NativeSandbox
import SwiftUI
import UIKit

@MainActor
enum LSDisplayStories {
    private static let groupLabel = "Atoms / Display"

    static let all: [Story] = [
        Story(
            id: "atoms.avatar.image",
            tier: .atom,
            component: "LSAvatar",
            name: "Avatar Image",
            summary: "\(groupLabel) image avatar."
        ) { _ in
            LSAvatar(image: UIImage(systemName: "person.fill"), size: .md)
        },
        Story(
            id: "atoms.avatar.initials",
            tier: .atom,
            component: "LSAvatar",
            name: "Avatar Initials",
            summary: "\(groupLabel) initials fallback."
        ) { _ in
            LSAvatar(initials: "JR", size: .md)
        },
        Story(
            id: "atoms.avatar.size-matrix",
            tier: .atom,
            component: "LSAvatar",
            name: "Avatar Size Matrix",
            summary: "\(groupLabel) size matrix."
        ) { _ in
            LSAvatarSizeMatrixStory()
        },
        Story(
            id: "atoms.divider.default",
            tier: .atom,
            component: "LSDivider",
            name: "Divider",
            summary: "\(groupLabel) subtle divider."
        ) { _ in
            LSDisplayDividerStory()
        },
        Story(
            id: "atoms.spinner.default",
            tier: .atom,
            component: "LSSpinner",
            name: "Spinner",
            summary: "\(groupLabel) native loading spinner."
        ) { _ in
            LSSpinner()
        },
    ]
}

private struct LSAvatarSizeMatrixStory: View {
    @Environment(\.theme) private var theme

    var body: some View {
        HStack(alignment: .center, spacing: theme.space.md) {
            ForEach(LSAvatar.Size.allCases, id: \.self) { size in
                VStack(spacing: theme.space.sm) {
                    LSAvatar(initials: label(for: size), size: size)
                    LSText(label(for: size), variant: .label.sm, color: .secondary)
                }
            }
        }
        .padding(theme.space.lg)
    }

    private func label(for size: LSAvatar.Size) -> String {
        switch size {
        case .xs:
            "XS"
        case .sm:
            "SM"
        case .md:
            "MD"
        case .lg:
            "LG"
        case .xl:
            "XL"
        }
    }
}

private struct LSDisplayDividerStory: View {
    @Environment(\.theme) private var theme

    var body: some View {
        VStack(alignment: .leading, spacing: theme.space.md) {
            LSText("Above", variant: .body.md)
            LSDivider()
            LSText("Below", variant: .body.md)
        }
        .padding(theme.space.lg)
    }
}

File: android/app/src/debug/java/com/laneshadow/sandbox/stories/LSDisplayStories.kt
package com.laneshadow.sandbox.stories

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.painter.ColorPainter
import com.laneshadow.theme.LocalLaneShadowTheme
import com.laneshadow.ui.atoms.AvatarSize
import com.laneshadow.ui.atoms.LSAvatar
import com.laneshadow.ui.atoms.LSDivider
import com.laneshadow.ui.atoms.LSSpinner
import com.nativesandbox.model.ComponentTier
import com.nativesandbox.model.Story

object LSDisplayStories {
    val all: List<Story> = listOf(
        Story(
            id = "atoms.avatar.image",
            tier = ComponentTier.Atom,
            component = "LSAvatar",
            name = "Avatar Image",
            summary = "Image-backed avatar clipped to a circular token-resolved frame.",
            content = { AvatarImageStory() },
        ),
        Story(
            id = "atoms.avatar.initials",
            tier = ComponentTier.Atom,
            component = "LSAvatar",
            name = "Avatar Initials",
            summary = "Initials fallback avatar rendered through LSText.",
            content = { AvatarInitialsStory() },
        ),
        Story(
            id = "atoms.avatar.size-matrix",
            tier = ComponentTier.Atom,
            component = "LSAvatar",
            name = "Avatar Sizes",
            summary = "All five avatar sizes using the live sizing ladder.",
            content = { AvatarSizesStory() },
        ),
        Story(
            id = "atoms.divider.default",
            tier = ComponentTier.Atom,
            component = "LSDivider",
            name = "Horizontal Divider",
            summary = "Hairline divider using the subtle border token mapping.",
            content = { DividerStory() },
        ),
        Story(
            id = "atoms.spinner.default",
            tier = ComponentTier.Atom,
            component = "LSSpinner",
            name = "Medium Spinner",
            summary = "Indeterminate spinner tinted with the default signal color.",
            content = { SpinnerStory() },
        ),
    )
}

@Composable
private fun AvatarImageStory() {
    val theme = LocalLaneShadowTheme.current

    Column(modifier = Modifier.padding(theme.space.lg)) {
        LSAvatar(
            image = ColorPainter(theme.colors.primary.default),
            size = AvatarSize.Md,
        )
    }
}

@Composable
private fun AvatarInitialsStory() {
    val theme = LocalLaneShadowTheme.current

    Column(modifier = Modifier.padding(theme.space.lg)) {
        LSAvatar(
            initials = "JR",
            size = AvatarSize.Md,
        )
    }
}

@Composable
private fun AvatarSizesStory() {
    val theme = LocalLaneShadowTheme.current

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(theme.space.lg),
        horizontalArrangement = Arrangement.spacedBy(theme.space.md),
    ) {
        AvatarSize.entries.forEach { size ->
            LSAvatar(initials = "JR", size = size)
        }
    }
}

@Composable
private fun DividerStory() {
    val theme = LocalLaneShadowTheme.current

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(theme.space.lg),
    ) {
        LSDivider()
    }
}

@Composable
private fun SpinnerStory() {
    val theme = LocalLaneShadowTheme.current

    Column(modifier = Modifier.padding(theme.space.lg)) {
        LSSpinner()
    }
}
